### PR TITLE
Update the TFJob image to the latest image and tag 0.3

### PR DESCRIPTION
**Pull request by jlewi**
* This makes the change on master
* Will need to cherry pick it onto the 0.3 branch afterwards.

* Related to #1599

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/1608)

### CODE REVIEW

File: docs_dev/releasing.md [hunk 1]
```diff
@@ -307,6 +310,35 @@ the correct Docker image.
...
+    * Add a version that specifies the sha of the image you want to use and the release
+      tag you want to add e.g. "vX.Y.Z"
+
+    ```
+
+    ```
+1. Run the following command to apply the new image tag
+
+   ```
+   releaseing/run_apply_image_tags.sh .*tf_operator.*:vX.Y.Z
```
Comment (richardsliu) [inline docs_dev/releasing.md]: Typo - "releasing"

**Comment by richardsliu** [comment]
Nit: In the future I think we should keep image updates from other changes in separate PRs. This makes it easier to revert if necessary.

**Comment by jlewi** [comment]
deploy_argo test failed;
/test all

**Pull request merged**